flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -206,6 +206,14 @@
 			key("task.cancellation.timeout")
 			.defaultValue(180000L);
 
+	/**
+	 * This configures how long we wait for the timers to finish all pending timer threads
+	 * when the stream task is cancelled .
+	 */
+	public static final ConfigOption<Long> TASK_CANCELLATION_TIMEOUT_TIMERS = ConfigOptions
+			.key("task.cancellation.timeout.timers")
+			.defaultValue(7500L);
```

Comment:
add deprecated key? (I'm not quite sure whether the previous option was part of a release)

Reply:
Good point - I just checked, it was in fact already part of 1.4 (I had falsely in my mind that it was new in 1.5)